### ISSUE TITLE
EDC 0.5.4 → Catalog Query Fails→ "PKIX path building failed"

### ISSUE THREAD

**Issue by jerzy-wachowiak** [eclipse-tractusx/tractusx-edc#1163]
EDC 0.5.4 Azure Deployment

**Catalog query fails ...**

- Call {{CONSUMER_MANAGEMENT_URL}}/catalog/request
```
curl --location 'https://bpnl00000007twdx.mgmt.edc.supplyon.com/management/v2/catalog/request' \
--header 'Content-Type: application/json' \
--header 'X-Api-Key: XXXXXXXXXXXXXXXXX' \
--data-raw '{
    "@context": {},
    "protocol": "dataspace-protocol-http",
    "counterPartyAddress": "https://bpnl00000007txd8.edc.supplyon.com/api/v1/dsp",
    "querySpec": {
        "offset": 0,
        "limit": 100,
        "filter": "",
        "range": {
            "from": 0,
            "to": 100
        },
        "criterion": ""
    }
}'
```
- Response:

```
[
    {
        "message": "org.eclipse.edc.spi.http.EdcHttpClientException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target",
        "type": "BadGateway",
        "path": null,
        "invalidValue": null
    }
]
```

- Data Consumer Log
Each try to call the catalog query generates log entry as below. The field in the call is named `counterPartyAddress` !
```

INFO 2024-03-26T14:33:13.92152681 edc054-tractusx-connector-azure-vault-controlplane ready
WARNING 2024-03-26T14:35:56.701743546 The attribute https://w3id.org/edc/v0.0.1/ns/providerUrl has been deprecated in type https://w3id.org/edc/v0.0.1/ns/CatalogRequest, please use https://w3id.org/edc/v0.0.1/ns/counterPartyAddress
WARNING 2024-03-26T14:39:56.082322118 The attribute https://w3id.org/edc/v0.0.1/ns/providerUrl has been deprecated in type https://w3id.org/edc/v0.0.1/ns/CatalogRequest, please use https://w3id.org/edc/v0.0.1/ns/counterPartyAddress
WARNING 2024-03-26T14:52:19.848825752 The attribute https://w3id.org/edc/v0.0.1/ns/providerUrl has been deprecated in type https://w3id.org/edc/v0.0.1/ns/CatalogRequest, please use https://w3id.org/edc/v0.0.1/ns/counterPartyAddress
WARNING 2024-03-26T14:54:17.596888394 The attribute https://w3id.org/edc/v0.0.1/ns/providerUrl has been deprecated in type https://w3id.org/edc/v0.0.1/ns/CatalogRequest, please use https://w3id.org/edc/v0.0.1/ns/counterPartyAddress
WARNING 2024-03-26T14:54:45.174427708 The attribute https://w3id.org/edc/v0.0.1/ns/providerUrl has been deprecated in type https://w3id.org/edc/v0.0.1/ns/CatalogRequest, please use https://w3id.org/edc/v0.0.1/ns/counterPartyAddress
```

**Comment by rafaelmag110** [comment]
Hey!
I'm operating a 0.5.4 Connector - but on an AWS environment - and found this issue as well in a very specific ocasion. It turned out to be that the CAs included in the SSL certificate of the server (the counterPartyAddress) could not be validated, as my client - my connector in this case - did not have the correct certificates installed. Please confirm the required CA certificates by the server can be resolved by the client JVM trust store.

**Comment by jerzy-wachowiak** [comment]
Thx, I will check this ... 😎

**Comment by rafaelmag110** [comment]
Hm, I see that to be a very challenging task, considering the team would need take into consideration various requirements from various entities. I don't believe that to be a scalable solution.

Usually this shouldn't be an issue, since most certificates should be signed by a widely trusted authority. There are only certain ocasion where this issue appear, and maybe yours is one of these ocasions.

With some online research you might be able to understand why and when this issue appears and how you can fix it for your specific case.

**Comment by jerzy-wachowiak** [comment]
Thx for your explanations! This is out of scope for us. The decision is up to the EDC Team.

**Comment by paullatzelsperger** [comment]
I fully agree with @rafaelmag110's assessment, and would like to add that - while obviously possible - I would caution against exposing connector endpoints directly to the internet, even protocol ones. 

In an actual real world scenario I would assume there to be at least an ingress controller or a load balancer, maybe even an API gateway of sorts, that typically takes care of SSL termination. However there are other concerns, such as rate-limiting, auditing etc. that need to be handled.

Even if we wanted to, we have no way of knowing ahead of time where people deploy EDC, what certificates they use, and what CAs' certificates need to be supported (in addition to the standard ones). We also somehow magically would have to know when certificates expire and must be renewed, etc.

Without devolving this into a philosophical discussion on what EDC is or isn't, I would like to point out that operational concerns like networking and security are outside the purview of EDC. 
To me, this sounds like a job for an operating company / SaaS provider.

**Comment by jerzy-wachowiak** [comment]
To make it clear. We use:
* Out-of-the box EDC Azure deployment in AKS (image, charts)
* No special configuration
* The problem hasn't been encountered neither in  EDC 0.5.3 nor in EDC 0.6.0

The ticket can be closed from my perspective as EDC 0.5.4 is a legacy connector.
